tmpfiles/src/lib.rs

```diff
@@ -18,7 +20,22 @@ use rustix::path::Arg;
 use thiserror::Error;
 
 const TMPFILESD: &str = "usr/lib/tmpfiles.d";
-const BOOTC_GENERATED: &str = "bootc-autogenerated-var.conf";
+/// The path to the file we use for generation
+const BOOTC_GENERATED_PREFIX: &str = "bootc-autogenerated-var";
+
+/// The number of times we've generated a tmpfiles.d
+#[derive(Debug, Default)]
+struct BootcTmpfilesGeneration(u32);
+
+impl BootcTmpfilesGeneration {
+    fn increment(&self) -> Self {
```

Comment:
Pedantic nit (take it or leave it), this should probably either consume `self` then hand it back, or take `&mut self` and do the increment in-place, just to avoid having multiple copies of the generation and accidentally using the wrong one.  It's fine right now since in usage you're always overwriting the old generation with the new instance but changing it would let the type system enforce that.

Worst case, derive `clone` and make it explicit that you intended to have two :smile:

Reply:
Good point, will do in a followup